### PR TITLE
HTTP2: Add GrpcNetworkStreamState metadata

### PR DESCRIPTION
**Pull request by yashykt**
HTTP2: Add GrpcNetworkStreamState metadata for calls that are not sent on wire and for those that are not seen by server

### CODE REVIEW

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 1]
```diff
@@ -1092,21 +1127,18 @@ void grpc_chttp2_add_incoming_goaway(grpc_chttp2_transport* t,
...
-  // cancel out streams that haven't yet started if we have received a GOAWAY
+  // maybe cancel out streams that haven't yet started if we have received a
+  // GOAWAY
   if (t->goaway_error != GRPC_ERROR_NONE) {
-    while (grpc_chttp2_list_pop_waiting_for_concurrency(t, &s)) {
-      grpc_chttp2_cancel_stream(
-          t, s,
-          grpc_error_set_int(
-              GRPC_ERROR_CREATE_FROM_STATIC_STRING("GOAWAY received"),
-              GRPC_ERROR_INT_GRPC_STATUS, GRPC_STATUS_UNAVAILABLE));
```
Comment (markdroth) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc:1108]: It looks like we're no longer setting the status to UNAVAILABLE here, but I see that the tests are still checking for this, and they are presumably passing, so it must be happening somewhere else.  Where is this happening now?

**Review by markdroth — APPROVED** [review]
This looks great to me, but please wait for Craig to review before merging.

Thanks for doing this!

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc (hunk 1 above)
Comment (yashykt) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc:1108]: https://github.com/grpc/grpc/blob/cf81e41162603d229d0c8ef05f137a0afbe1f3c6/src/core/ext/transport/chttp2/transport/chttp2_transport.cc#L1054

goaway_error sets the status already

Comment (markdroth) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc:1108]: Got it, thanks!

**Review by ctiller — APPROVED** [review]

**Pull request merged**